### PR TITLE
Cvm 20220317

### PR DESCRIPTION
**Pull request by zhaocaibei123**
### PR types

New features
### PR changes

Others
### Describe

Paddle distributed framework support two-phase training for ps async mode now, and we will provide two-phase training demo in PaddleRec later.
1. move BrpcPSClient from Communicator to FleetWrapper, remove Communicator for ps async mode 
2. add DownpourLiteWorker to support more function than HogwilderWorker, like metric and others
3. add SparseAccessor for common embedding, and modify CommonCtrAccessor for embedding with show/click
4. make new compile-time refactor for programs takes into effect, and fix some bugs

### CODE REVIEW

File: paddle/fluid/distributed/ps/service/brpc_ps_client.cc [hunk 1]
```diff
@@ -238,7 +238,7 @@ int32_t BrpcPsClient::initialize() {
       std::thread(std::bind(&BrpcPsClient::push_dense_task_consume, this));
   // for debug
   // _print_thread =
```
Comment (fuyinno4) [inline paddle/fluid/distributed/ps/service/brpc_ps_client.cc:240]: 没用的代码删掉？

File: paddle/fluid/distributed/ps/table/sparse_accessor.cc [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+
+#include "paddle/fluid/distributed/ps/table/sparse_accessor.h"
+#include <gflags/gflags.h>
+#include "glog/logging.h"
+#include "paddle/fluid/string/string_helper.h"
+
+namespace paddle {
+namespace distributed {
+
+int SparseAccessor::initialize() {
```
Comment (fuyinno4) [inline paddle/fluid/distributed/ps/table/sparse_accessor.cc:23]: 可以备注下// no show click, for word2vec(DownpourSparseValueAccessor )

File: paddle/fluid/framework/device_worker.h [hunk 3]
```diff
@@ -341,6 +349,78 @@ class DownpourWorker : public HogwildWorker {
   // std::vector<std::pair<uint64_t, uint64_t>> copy_dense_tables_;
 };
 
+#if defined(PADDLE_WITH_PSCORE)
+class DownpourLiteWorker : public HogwildWorker {
```
Comment (fuyinno4) [inline paddle/fluid/framework/device_worker.h:353]: 加注释：Based on DownpourWorker，remove push pull code into operator

File: paddle/fluid/framework/pull_dense_worker.cc [hunk 4]
```diff
@@ -61,7 +61,13 @@ void PullDenseWorker::Initialize(const TrainerDesc& param) {
     last_versions_[tid] = 0;
     current_version_[tid] = 0;
   }
+
   fleet_ptr_ = FleetWrapper::GetInstance();
+
+#if defined(PADDLE_WITH_PSCORE)
+  new_fleet_ptr_ = paddle::distributed::FleetWrapper::GetInstance();
```
Comment (fuyinno4) [inline paddle/fluid/framework/pull_dense_worker.cc:68]: 直接用fleet_ptr_ = xxx?

File: python/paddle/distributed/ps/the_one_ps.py [hunk 5]
```diff
@@ -119,11 +133,17 @@ def _set(self, accessor_proto, varname, program_id, context):
                 break
 
         if not accessor_proto.HasField("accessor_class"):
-            accessor_proto.accessor_class = "CtrCommonAccessor"
```
Comment (fuyinno4) [inline python/paddle/distributed/ps/the_one_ps.py:122]: 备注下：DownpourSparseValueAccessor  DownpourUnitAccessor

File: paddle/fluid/distributed/ps/table/ctr_accessor.cc [hunk 6]
```diff
@@ -55,7 +55,7 @@ size_t CtrCommonAccessor::mf_size() {
 // pull value
 size_t CtrCommonAccessor::select_dim() {
```
Comment (fuyinno4) [inline paddle/fluid/distributed/ps/table/ctr_accessor.cc:66]: 备注下: DownpourUnitAccessor

**Review by yaoxuefeng6 — DISMISSED** [review]
LGTM

**Review by fuyinno4 — APPROVED** [review]

**Pull request closed without merging**